src/components/NetworkMonitor.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { notify, removeNotification } from '../actions'
+import { fetchJSON } from '../util/api'
+
+@connect(() => ({}), {notify, removeNotification})
+export default class NetworkMonitor extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {}
+  }
+
+  render () {
+    return null
+  }
+
+  componentDidMount () {
+    this.visibility = require('visibility')()
+    setInterval(() => {
+      if (!this.visibility.visible()) return
+
+      fetchJSON('/noo/user/status')
+      .then(() => {
+        if (this.state.offline) {
+          this.props.removeNotification('offline')
```

Comment:
👍 Yup! This is super helpful going forward for testing and generally decoupling component crafting from app state management...